src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -2,58 +2,244 @@
+                    }
+
+                    // Otherwise, we redirect user to the specified module, variable, or resource declaration
+                    if (GetObjectSyntaxFromDeclaration(ancestorSymbol.DeclaringSyntax) is ObjectSyntax objectSyntax
+                        && ObjectSyntaxExtensions.SafeGetPropertyByNameRecursive(objectSyntax, propertyAccesses.ToArray()) is ObjectPropertySyntax resultingSyntax)
+                    {
+                        return Task.FromResult(new LocationOrLocationLinks(new LocationOrLocationLink(new LocationLink
+                        {
+                            OriginSelectionRange = result.Origin.ToRange(result.Context.LineStarts),
+                            TargetUri = request.TextDocument.Uri,
+                            TargetRange = resultingSyntax.ToRange(result.Context.LineStarts),
+                            TargetSelectionRange = resultingSyntax.ToRange(result.Context.LineStarts)
+                        })));
+                    }
+                }
                 return Task.FromResult(new LocationOrLocationLinks());
             }
 
-            if (result.Symbol is DeclaredSymbol declaration)
+            return Task.FromResult(new LocationOrLocationLinks());
+        }
+
+        private Task<LocationOrLocationLinks> GetModuleSymbolLocationAsync(
+            DefinitionParams request,
+            SymbolResolutionResult result,
+            CompilationContext context,
+            ModuleDeclarationSyntax moduleDeclarationSyntax,
+            string propertyType,
+            string propertyName)
+        {
+            if (moduleDeclarationSyntax.TryGetPath() is StringSyntax pathSyntax
+            && pathSyntax.StringTokens is ImmutableArray<Token> stringTokens
+            && stringTokens.Length == 1
+            && stringTokens[0] is Token stringToken
+            && stringToken.Type == TokenType.StringComplete
+            && fileResolver.TryResolveFilePath(request.TextDocument.Uri.ToUri(), stringToken.Text.Replace("'", string.Empty)) is Uri moduleUri
+            && fileResolver.TryFileExists(moduleUri))
             {
-                return Task.FromResult(new LocationOrLocationLinks(new LocationOrLocationLink(new LocationLink
+                CompilationContext? moduleContext = null;
+                // if the file's been opened in the workspace, we should be able to get its compilation
+                if (this.compilationManager.GetCompilation(DocumentUri.From(moduleUri)) is CompilationContext existingModuleContext)
                 {
-                    // source of the link
-                    OriginSelectionRange = result.Origin.ToRange(result.Context.LineStarts),
-                    TargetUri = request.TextDocument.Uri,
+                    moduleContext = existingModuleContext;
+                }
+                // otherwise, we have to add it manually
+                else if (fileResolver.TryRead(moduleUri, out var fileContents, out var _))
```

Comment:
Not sure how else to get a compilation of a module Uri that hasn't been opened in editor yet (that's when it gets added by itself). Currently, if the file is edited under our noses (e.g. vim) then we will not upsert the changes,

Reply:
Would you be able to get it from `IWorkspace` instead?

Reply:
Thanks for the suggestion! I was able to dig through and find that most this work is done, @shenglol 's recent work `Compilation.SourceFileGrouping.LookUpModuleSourceFile(moduleDeclarationSyntax)` made this really easy :)